MapleServer2/Packets/TrophyPacket.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Types;
+
+namespace MapleServer2.Packets
+{
+    public static class TrophyPacket
+    {
+        private enum TrophyPacketMode : byte
+        {
+            TableStart = 0x0,
+            TableContent = 0x1,
+            Update = 0x2
+        }
+
+        public enum GradeStatus : byte
+        {
+            NotFinalGrade = 0x0,
+            FinalGrade = 0x3
+        }
+
+        public static Packet WriteTableStart()
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.ACHIEVE);
+            pWriter.WriteEnum(TrophyPacketMode.TableStart);
+
+            return pWriter;
+        }
+
+        // packet from WriteTableStart() must be sent immediately before sending these packets
+        public static Packet WriteTableContent(List<Trophy> trophies)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.ACHIEVE);
+            pWriter.WriteEnum(TrophyPacketMode.TableContent);
+            pWriter.WriteInt(trophies.Count);
+
+            foreach (Trophy trophy in trophies)
+            {
+                pWriter.WriteInt(trophy.Id);
+                pWriter.WriteInt(1);            // unknown 'SS' ?
+                WriteIndividualTrophy(pWriter, trophy);
+            }
+
+            return pWriter;
+        }
+
+        public static Packet WriteUpdate(Trophy trophy)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.ACHIEVE);
```

Comment:
Maybe rename the SendOp and RecvOp too, but I'm not sure about that.

Reply:
yeah coz i think those are the same as the ones in MapleShark2 so I might keep them that way for now

Reply:
Yes but it's like the discussion between ingame and xml names. We should use the ingame name always right? So between ingame and mapleshark I would prefer to use ingame terminology

Reply:
i see that other opcodes are named that way as well (ex. STAT for StatPacket.cs). Modified to follow convention.